packages/shortest/src/browser/core/browser-tool.ts

```diff
@@ -875,4 +875,84 @@ export class BrowserTool extends BaseBrowserTool {
       },
     );
   }
+
+  /**
+   * Waits for DOM to stabilize
+   * DOM is considered stabilized when:
+   * - DOMContentLoaded is fired
+   * - No mutations are detected for 1 second (e.g new elements such as modals, popups, etc.)
```

Comment:
What if new elements are shown after more than 1 second? Or once the page scrolling is used?